src/core/server/http/http_tools.ts

```diff
@@ -169,3 +170,12 @@ export function defaultValidationErrorHandler(
 
   throw err;
 }
+
+export function getRequestId(request: Request, options: HttpConfig['requestId']): string {
+  return options.allowFromAnyIp ||
+    // socket may be undefined in integration tests that connect via the http listener directly
+    (request.raw.req.socket?.remoteAddress &&
+      options.ipAllowlist.includes(request.raw.req.socket.remoteAddress))
+    ? request.headers['x-opaque-id'] ?? uuid.v4()
+    : uuid.v4();
```

Comment:
Can we extract the id generation logic in a common place to use it here and in KibanaRequest contructor? 
``js
export function generateId() {
  return uuid.v4();
}
```